frontends/p4/fromv1.0/programStructure.cpp

```diff
@@ -405,14 +413,39 @@ const IR::ParserState* ProgramStructure::convertParser(const IR::V1Parser* parse
             auto deststate = getState(state);
             if (deststate == nullptr)
                 return nullptr;
+
+            // discover all parser value_sets in the current parser state.
+            for (auto v : c->values) {
+                auto first = v.first->to<IR::PathExpression>();
+                if (!first) continue;
+                auto value_set = value_sets.get(first->path->name);
+                if (!value_set) {
+                    ::error("Unable to find declaration for parser_value_set %s", first->path->name);
+                    return nullptr;
+                }
+
+                auto type = new IR::Type_Name("value_set");
+                unsigned vset_size = 4;  // default value_set size is 4.
+                if (auto size = value_set->getAnnotation("parser_value_set_size")) {
```

Comment:
If this annotation is present, you probably want to remove it, as the size is now a ctor param...